### ISSUE TITLE
Support link IDs

### ISSUE THREAD

**Issue by wheeOs** [mermaid-js/mermaid#4568]
### Proposal

It should be optionally possible to define IDs for links. This will make it it easier to reference links in `linkStyle`. Currently afaics specific links can only be referenced by index. Imagine a diagram where a lot of links are styled in the current way. You only have to remove or add a link prior to those styled links and all of those styling directives get invalid since the indexes of the targeted links have changed naturally. ID or key references in linkStyle would prevent this inconvenience.

### Example

```
A L-->|link label| B
B "link id with whitespaces"--> C

linkStyle L stroke-width: 3px, stroke:blue
linkStyle "link id with whitespaces" stroke-width: 1px, stroke:red
```

**Comment by sidharthv96** [comment]
Would this solve your issue?

https://github.com/mermaid-js/mermaid/pull/4483

**Comment by wheeOs** [comment]
@sidharthv96 yes that would be even a better solution in cases of "inlining" (writing the linkStyle close to the link line. But still, I would suggest the ID approach when someone prefers to write style statements in a separate block, similar to my given example.

**Comment by sidharthv96** [comment]
The only issue I see is that the syntax becomes more complicated, but as it's optional, we could have it if there's enough interest from the community. 
IDs with whitespaces are not supported anywhere else AFAIK, so we can remove that from the proposal, which simplifies it.

**Comment by rockmasterflex69** [comment]
I'm not sure about whitespace being super critical, but being able to pre-assign (even numeric only) IDs to a link is crucial to dynamic graph styling. Is that still up for discussion here?

**Comment by vincentor** [comment]
I really need this feature too, in my scenario, indexes are difficult to calculate because links are generating dynamically, in particular flow charts with subgraphs

**Comment by phil294** [comment]
This seems to be a dupe of #4794

Would be fantastic to have either way! (or #4483, for that matter)

**Comment by simplenotezy** [comment]
Agreed - this would be a big plus!
